utils/common.py

```diff
@@ -53,9 +53,9 @@ def wrapper(*args,**kwargs):
        
        user=users.getUser(uid)
        
-       has_access=(user is not None) and (hash==user.password_hash)
-       
-       if not has_access:
+       has_access=(user is not None) and (hash==user.password_hash) 
```

Comment:
Query to get number of warnings and time of last warning --- don't use bespoke fields.